### PR TITLE
added addon delete function to addons api

### LINKED ISSUE #19072: enable addon delete via addons api
You can submit, edit, and add new versions via the addons api; we should implement delete also.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #19072

### CODE REVIEW

File: src/olympia/addons/utils.py [hunk 1]
```diff
@@ -409,3 +415,33 @@ def fetch_translations_from_addon(addon, properties):
...
     return {str(value) for value in Translation.objects.filter(id__in=translation_ids)}
+
+
+def generate_delete_token(addon_id):
+    token_payload = {
+        'exp': datetime.utcnow() + timedelta(seconds=60),
+        'addon_id': addon_id,
+        'action': 'delete',
+    }
+    return jwt.encode(token_payload, settings.SECRET_KEY)
```
Comment (eviljeff) [inline src/olympia/addons/utils.py:426]: I think this was what you had in mind around a special token?

Comment (diox) [reply src/olympia/addons/utils.py:426]: Yes, but let's not use `jwt` for this. Use django's `sign`: https://docs.djangoproject.com/en/4.0/topics/signing/

Comment (eviljeff) [reply src/olympia/addons/utils.py]: done https://github.com/mozilla/addons-server/pull/19147/commits/0597956cbe27508c7cd53a66d89c6886b2d3628a

File: src/olympia/addons/models.py [hunk 2]
```diff
@@ -2057,7 +2057,7 @@ def get_queryset(self):
 class AddonUser(OnChangeMixin, SaveUpdateMixin, models.Model):
     id = PositiveAutoField(primary_key=True)
     addon = models.ForeignKey(Addon, on_delete=models.CASCADE)
-    user = user = models.ForeignKey(UserProfile, on_delete=models.CASCADE)
```
Comment (diox) [inline src/olympia/addons/models.py:2060]: :see_no_evil:

File: docs/topics/api/addons.rst [hunk 3]
```diff
@@ -381,6 +381,47 @@ Note: as form-data can not include objects, and creating an add-on requires the
...
+    :query string delete_confirm: the confirmation token from the :ref:`delete confirm <addon-delete-confirm>` endpoint.
+
+
+~~~~~~~~~~~~~~
+Delete Confirm
+~~~~~~~~~~~~~~
+
+.. _addon-delete-confirm:
+
+This endpoint just supplies a special signed JWT token that can be used to confirm deletion of an add-on.
```
Comment (diox) [inline docs/topics/api/addons.rst:411]:
```suggestion
This endpoint just supplies a special signed token that can be used to confirm deletion of an add-on.
```

**Review by diox — APPROVED** [review]

**Pull request merged**